Automattic/wp-calypso
# Inject Calypso's domain contact details form into composite checkout

#### Changes proposed in this Pull Request

* Inject `ContactDetailsFormFields` as a managed component into composite checkout.

#### Testing instructions

Tests: `npm install && npm run test-packages composite-checkout-wpcom && npm run test-client client/my-sites/checkout/checkout`
Typecheck: `cd packages/composite-checkout-wpcom/ && tsc; cd ../..`

For manual testing:
1. Start calypso with composite checkout enabled using `ENABLE_FEATURES=composite-checkout-wpcom npm start`
2. Make sure you have the store and billing sandboxes enabled on the backend
3. Enter checkout with a domain in your cart.
4. Proceed to the contact details form. Fill out the form, checking that the `Continue` button does not become active until the form is complete. Watch the console for validation errors.
5. In the review step, delete the domain from your cart.
6. Return to the contact details step; verify that the fields there are already filled in.
